Add itchio/butler

Cantido opened the pull request:
[itchio/butler](https://github.com/itchio/butler) - Command-line itch.io helper

## Check List

<!-- Please check the list. Please don't remove the check list. -->

- [X] Read [CONTRIBUTING.md](https://aquaproj.github.io/docs/products/aqua-registry/contributing)
  - [X] Read [OSS Contribution Guide](https://github.com/suzuki-shunsuke/oss-contribution-guide/blob/main/README.md)
- [X] [All commits are signed](https://github.com/suzuki-shunsuke/oss-contribution-guide/blob/main/docs/commit-signing.md)
  - This repository enables `Require signed commits`, so all commits must be signed
- [X] [Avoid force push](https://github.com/suzuki-shunsuke/oss-contribution-guide?tab=readme-ov-file#dont-do-force-pushes-after-opening-pull-requests)
- [ ] Install and execute the package and confirm if the package works well
- [Execute `cmdx s` to scaffold code](https://aquaproj.github.io/docs/products/aqua-registry/contributing/#use-cmdx-s-definitely)

<!-- Please write the description here -->

Butler is a CLI tool for interacting with the games platform itch.io. The tool is distributed via this website: https://broth.itch.zone/butler. The downloaded zips contain the executable and two library files.

If possible, I'd like a hand with this, because I'm having trouble debugging it. It looks like aqua is successfully getting and extracting the zip, but when I try to run it in the test container (and on my local machine), I get an error. Here's the full output of `aqua --log-level debug exec butler` run on my local (Arch linux) machine:

```
DEBU[0000] no version_constraint matches                 env=linux/amd64 package_version=0.12 program=aqua program_version=2.53.2 version_constraint=false
DEBU[0000] reading registry cache                        config_file_path=/home/rosa/workspace/aqua-registry/aqua.yaml env=linux/amd64 exe_name=butler program=aqua program_version=2.53.2
DEBU[0000] read a registry cache file                    config_file_path=/home/rosa/workspace/aqua-registry/aqua.yaml env=linux/amd64 error=open a registry cache: open /home/rosa/.local/share/aquaproj-aqua/registry-cache/L2hvbWUvcm9zYS93b3Jrc3BhY2UvYXF1YS1yZWdpc3RyeS9hcXVhLnlhbWw=.json: no such file or directory exe_name=butler program=aqua program_version=2.53.2
DEBU[0000] getting a package from a registry             config_file_path=/home/rosa/workspace/aqua-registry/aqua.yaml env=linux/amd64 exe_name=butler package_name=aquaproj/registry-tool program=aqua program_version=2.53.2 registry_name=standard
DEBU[0000] match the version_constraint                  config_file_path=/home/rosa/workspace/aqua-registry/aqua.yaml env=linux/amd64 exe_name=butler package_name=aquaproj/registry-tool package_semver=v0.3.5 package_version=v0.3.5 program=aqua program_version=2.53.2 registry_name=standard version_constraint=semver(">= 0.1.6")
DEBU[0000] getting a package from a registry             config_file_path=/home/rosa/workspace/aqua-registry/aqua.yaml env=linux/amd64 exe_name=butler package_name=rhysd/actionlint program=aqua program_version=2.53.2 registry_name=standard
DEBU[0000] match the version_constraint                  config_file_path=/home/rosa/workspace/aqua-registry/aqua.yaml env=linux/amd64 exe_name=butler package_name=rhysd/actionlint package_semver=v1.7.7 package_version=v1.7.7 program=aqua program_version=2.53.2 registry_name=standard version_constraint=true
DEBU[0000] getting a package from a registry             config_file_path=/home/rosa/workspace/aqua-registry/aqua.yaml env=linux/amd64 exe_name=butler package_name=suzuki-shunsuke/cmdx program=aqua program_version=2.53.2 registry_name=standard
DEBU[0000] match the version_constraint                  config_file_path=/home/rosa/workspace/aqua-registry/aqua.yaml env=linux/amd64 exe_name=butler package_name=suzuki-shunsuke/cmdx package_semver=v2.0.1 package_version=v2.0.1 program=aqua program_version=2.53.2 registry_name=standard version_constraint=true
DEBU[0000] getting a package from a registry             config_file_path=/home/rosa/workspace/aqua-registry/aqua.yaml env=linux/amd64 exe_name=butler package_name=jqlang/jq program=aqua program_version=2.53.2 registry_name=standard
DEBU[0000] match the version_constraint                  config_file_path=/home/rosa/workspace/aqua-registry/aqua.yaml env=linux/amd64 exe_name=butler package_name=jqlang/jq package_semver=1.8.1 package_version=jq-1.8.1 program=aqua program_version=2.53.2 registry_name=standard version_constraint=true
DEBU[0000] getting a package from a registry             config_file_path=/home/rosa/workspace/aqua-registry/aqua.yaml env=linux/amd64 exe_name=butler package_name=cli/cli program=aqua program_version=2.53.2 registry_name=standard
DEBU[0000] match the version_constraint                  config_file_path=/home/rosa/workspace/aqua-registry/aqua.yaml env=linux/amd64 exe_name=butler package_name=cli/cli package_semver=v2.74.2 package_version=v2.74.2 program=aqua program_version=2.53.2 registry_name=standard version_constraint=true
DEBU[0000] getting a package from a registry             config_file_path=/home/rosa/workspace/aqua-registry/aqua.yaml env=linux/amd64 exe_name=butler package_name=itchio/butler program=aqua program_version=2.53.2 registry_name=standard
DEBU[0000] no version_constraint                         config_file_path=/home/rosa/workspace/aqua-registry/aqua.yaml env=linux/amd64 exe_name=butler package_name=itchio/butler program=aqua program_version=2.53.2 registry_name=standard
DEBU[0000] updating registry cache                       config_file_path=/home/rosa/workspace/aqua-registry/aqua.yaml env=linux/amd64 exe_name=butler program=aqua program_version=2.53.2
DEBU[0000] installing the package                        env=linux/amd64 exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2 registry=standard
DEBU[0000] check if the package is already installed     env=linux/amd64 exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2 registry=standard
DEBU[0000] check the permission                          env=linux/amd64 exe_name=butler file_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2 registry=standard
DEBU[0000] check if exec file exists                     env=linux/amd64 exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2
DEBU[0000] execute the command                           env=linux/amd64 exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2
DEBU[0000] the process isn't started. retry              env=linux/amd64 error=call execve(2): exec format error exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2 retry_count=1
DEBU[0000] execute the command                           env=linux/amd64 exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2
DEBU[0000] the process isn't started. retry              env=linux/amd64 error=call execve(2): exec format error exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2 retry_count=2
DEBU[0000] execute the command                           env=linux/amd64 exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2
DEBU[0000] the process isn't started. retry              env=linux/amd64 error=call execve(2): exec format error exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2 retry_count=3
DEBU[0000] execute the command                           env=linux/amd64 exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2
DEBU[0000] the process isn't started. retry              env=linux/amd64 error=call execve(2): exec format error exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2 retry_count=4
DEBU[0000] execute the command                           env=linux/amd64 exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2
DEBU[0000] the process isn't started. retry              env=linux/amd64 error=call execve(2): exec format error exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2 retry_count=5
DEBU[0000] execute the command                           env=linux/amd64 exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2
DEBU[0000] the process isn't started. retry              env=linux/amd64 error=call execve(2): exec format error exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2 retry_count=6
DEBU[0000] execute the command                           env=linux/amd64 exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2
DEBU[0000] the process isn't started. retry              env=linux/amd64 error=call execve(2): exec format error exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2 retry_count=7
DEBU[0000] execute the command                           env=linux/amd64 exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2
DEBU[0000] the process isn't started. retry              env=linux/amd64 error=call execve(2): exec format error exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2 retry_count=8
DEBU[0000] execute the command                           env=linux/amd64 exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2
DEBU[0000] the process isn't started. retry              env=linux/amd64 error=call execve(2): exec format error exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2 retry_count=9
DEBU[0000] execute the command                           env=linux/amd64 exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2
DEBU[0000] the process isn't started. retry              env=linux/amd64 error=call execve(2): exec format error exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2 retry_count=10
FATA[0000] aqua failed                                   env=linux/amd64 error=it failed to start the process exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2
```

Cantido committed:
feat(itchio/butler): scaffold itchio/butler

Cantido committed:
use HTTP for itchio/butler

Cantido committed:
Merge branch 'main' into feat/itchio/butler

suzuki-shunsuke commented:
Thank you for your contribution!

📝

https://broth.itch.zone/butler
https://broth.itch.zone/butler/darwin-amd64
https://broth.itch.zone/butler/darwin-amd64/15.24.0

suzuki-shunsuke committed:
fix(itchio/butler): fix config

- Use `rosetta2`, `windows_arm_emulation`
- Use `append_ext`
- Fix supported_envs
- Use `version_constraint` and `version_overrides`

suzuki-shunsuke commented:
```
DEBU[0000] execute the command                           env=linux/amd64 exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2
DEBU[0000] the process isn't started. retry              env=linux/amd64 error=call execve(2): exec format error exe_name=butler package_name=itchio/butler package_version=15.24.0 program=aqua program_version=2.53.2 retry_count=1
```

I resolved this error.
This error occurred because you tried to execute a zip file.
I fixed the format to `zip`, then aqua unarchives the zip file. 5181ee1b6c72ef64099359021c1647e5ddbbced2

suzuki-shunsuke committed:
fix(itchio/butler): add "v" to versions

suzuki-shunsuke commented:
I added the prefix `v` to versions according to the GitHub Releases. ac147ec76cee0424735e82d4adbb213072078706

https://github.com/itchio/butler/releases

suzuki-shunsuke committed:
Merge branch 'main' into feat/itchio/butler

suzuki-shunsuke commented:
🎉 https://github.com/aquaproj/aqua-registry/releases/tag/v4.383.0

Cantido commented:
Thank you so much!
